Ejercicios resueltos

Alfred021 opened the pull request:
Ejercicios resueltos con todos los tests completados.

Alfred021 committed:
Ejercicios resueltos

Ejercicios resueltos con todos los tests completados.

jabsatz commented on src/ejercicios.js:
```diff
@@ -1,21 +1,44 @@
+    } else if (n < 2) {
+        return n
+    } else {
+        cache[n] = fibo(n - 1, cache) + fibo(n-2, cache)
```
Ojo con mutar cache acá, una forma más correcta sería
```
const newCache = [...cache]
newCache[n] = fibo(n - 1, cache) + fibo(n-2, cache)
```

Alfred021 replied:
El tema es que si lo implemento así nunca finaliza el ultimo caso de los tests:

` // si la función no termina, están calculando el resultado de una manera muy poco eficiente.
    expect(fns.fibo(100)).toEqual(354224848179262000000);`

Alfred021 replied:
@jabsatz Ahi le encontre una vuelta

```
const fibo = function() {
    let cache = [];
    return function fib(n) {
    if (cache[n]) {
        return cache[n]
    } else if (n < 2) {
        return n
    } else {
        cache[n] = fib(n - 1) + fib(n - 2)
    }
    return cache[n];
  }
}()
```
Con el () llamo a la funcion fibo inmediamente, eso devuelve la funcion que regresa y cuando lo llamas en el test le pasas el numero directamente como parametro, y esa funcion que devuelve es la que es llamada en cada recursion.

jabsatz commented on src/ejercicios.js:
```diff
@@ -1,21 +1,44 @@
 
-export const multiplicarAtributo = '???';
+export const multiplicarAtributo = fp.curry((attr, obj) => {
+    return obj[attr].reduce((acc, n) => acc * n);
```
está bien, pero fijate que estás repitiendo el código de `atributo` y `multiplicacion`. podés directamente usar esas funciones

jabsatz commented on src/ejercicios.js:
```diff
@@ -1,21 +1,44 @@
 
-export const mayorPersona = '???';
+export const mayorPersona = arr => {
+    const firstElement = fp.first(fp.reverse(fp.sortBy(element => element['edad'], arr)))
```
lo mismo acá, podés usar `ordenarPor` y `atributo`

Alfred021 committed:
Correcciones

Correcciones realizadas a 3 funciones siguiendo los lineamientos recibidos en la review
